classes/xt_quircks.bbclass

```diff
@@ -0,0 +1,58 @@
+def get_tarfname(location, d):
+    return os.path.join(os.path.dirname(location),
+        str(d.getVar('PN')) + "-" + str(d.getVar('PV')) + "-" +
+        os.path.basename(location) + ".tar.gz")
+
+def create_tarfile(location, d):
+    import tarfile
+
+    arcname = get_tarfname(location, d)
+    tar = tarfile.open(arcname, "w:gz")
+    tar.add(location, os.path.basename(location))
+    tar.close()
+    return arcname
+
+def check_url_or_pack(url, type, location, d):
+    if type.lower() != 'file':
+        return url
+    if not os.path.isdir(location):
+        return url
+    new_location = create_tarfile(location, d)
+    return url.replace(location, new_location)
+
+def remove_tarfile(url, type, location, d):
+    if type.lower() != 'file':
+        return url
+    if not os.path.isdir(location):
+        return url
+    arcname = get_tarfname(location, d)
+    if os.path.exists(arcname):
+        os.remove(arcname)
+
+addtask unpack_xt_extras after do_unpack before do_configure
+python do_unpack_xt_extras() {
+    bb.debug(1, "Unpacking Xen-troops extras")
+    # all the sources in SRC_URI have been unpacked already, clean it
+    d.setVar("SRC_URI", "")
```

Comment:
as per my understanding Yocto/bitbake do not use global environment, e.g. the "d" variable, which represents current execution thread own environment has no visibility to other's threads env and vise versa. Thus, I consider this safe.
What is more, this "d" variable changes do not even survive between different tasks of the same recipe.
Won't fix